frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineResourceDropdown.tsx

```diff
@@ -41,21 +42,27 @@ const PipelineResourceDropdown: React.FC<PipelineResourceDropdownProps> = (props
     resourceDefinition,
   );
 
-  const availableResources: PipelineResourceKind[] = resources.filter(
-    (resource) => resource.spec.type === filterType,
+  const availableResources: PipelineResourceKind[] = React.useMemo(
+    () => resources.filter((resource) => resource.spec.type === filterType),
+    [filterType, resources],
   );
 
-  const canAutoSelect = autoSelect && !touched && loaded && !error;
+  const canAutoSelect = autoSelect && !touched && loaded && !error && !!pipelineName;
   React.useEffect(() => {
     if (canAutoSelect) {
-      if (availableResources.length === 0) {
-        setFieldValue(name, CREATE_PIPELINE_RESOURCE);
+      const matchingResources = availableResources.filter(
+        (resource) => resource.metadata?.labels?.['tekton.dev/pipeline'] === pipelineName,
+      );
+      if (matchingResources.length > 0) {
+        setFieldValue(name, matchingResources[0].metadata.name, false);
+      } else if (availableResources.length > 0) {
+        setFieldValue(name, availableResources[0].metadata.name, false);
       } else {
-        setFieldValue(name, availableResources[0].metadata.name);
+        setFieldValue(name, CREATE_PIPELINE_RESOURCE, false);
```

Comment:
If we drop this else-case as well the user will show a "Select Resource" instead of seeing the option "Create resource" with an additional input field. So my recommendation is to set the value only if we found a matching resource.